crates/matrix-sdk-ui/src/spaces/mod.rs

```diff
@@ -211,6 +211,33 @@ impl SpaceService {
         spaces
     }
 
+    /// Returns a flattened list containing all joined spaces where the
+    /// user has permission to send `m.space.child` state events.
+    ///
+    /// Note: Unlike [`Self::joined_spaces()`], this method does not recompute
+    /// graph, nor does it notify subscribers about changes.
+    pub async fn all_editable_joined_spaces(&self) -> Vec<SpaceRoom> {
```

Comment:
One way to make this name a bit more pleasant is to remove the `all` prefix, and the joined nominator.

I think it's a given that you can only edit joined spaces or rooms for that matter.

So `editable_spaces()` sounds more palatable to me.

Reply:
> I think it's a given that you can only edit joined spaces or rooms for that matter.

Oh yeah good point, totally missed that fact 😆